lpcompiler.pas

```diff
@@ -1342,12 +1344,17 @@ function TLapeCompiler.ParseMethodHeader(out Name: lpString; addToScope: Boolean
         begin
           if addToScope then
             FStackInfo.addSelfVar(Lape_SelfParam, TLapeType(Typ));
+
+          methodDef := Result.MethodDef;
           Result.Free();
           Result := TLapeType_MethodOfType.Create(Self, TLapeType(Typ), nil, nil, '', @Pos);
+          Result.MethodDef := methodDef;
         end
         else if (not (Typ is TLapeType_SystemUnit)) then
           LapeException(lpeTypeExpected, [Tokenizer]);
-      end;
+      end
+      else if (Result.MethodDef = mdProperty) then
+        LapeException(lpeExpectedProperty, Tokenizer.DocPos);
```

Comment:
I will then argue that type properties is known from object pascal. Somewhat neater there as it's in the type declaration, but we wont wtf to the same degree when you encounter `Type.Prop := ..` because the syntax is recognizable from object psacal for anyone that has experience with it.